cmd/controller/app/controller.go

```diff
@@ -0,0 +1,87 @@
+/*
```

Comment:
I believe the threshold for distinguishing a move with modification from a delete and an add is that the modification has to touch less than 50% of the file's lines.  I don't have an official citation for this offhand, but there is some kind of similarity threshold.